src/Button.php

```diff
@@ -4,95 +4,425 @@
+     */
+    public function addClass(string $value): self
+    {
+        $new = clone $this;
+        $new->cssClass[] = $value;
+
+        return $new;
+    }
+
+    /**
+     * Adds a style class for the button component.
+     *
+     * @param array|string $value The style class for the button component. If an array, the values will be separated by
+     * a space. If a string, it will be added as is. For example, 'color: red;'. If the value is an array, the values
+     * will be separated by a space. e.g., ['color' => 'red', 'font-weight' => 'bold'] will be rendered as
+     * 'color: red; font-weight: bold;'.
+     * @param bool $overwrite Whether to overwrite existing styles with the same name. If `false`, the new value will be
+     * appended to the existing one.
+     *
+     * @return self A new instance with the specified style class value added.
+     */
+    public function addCssStyle(array|string $value, bool $overwrite = true): self
+    {
+        $new = clone $this;
+        Html::addCssStyle($new->attributes, $value, $overwrite);
+
+        return $new;
+    }
+
+    /**
+     * Sets the 'aria-expanded' attribute for the button, indicating whether the element is currently expanded or
+     * collapsed.
+     *
+     * @param bool $value The value to set for the 'aria-expanded' attribute.
+     *
+     * @return self A new instance with the specified 'aria-expanded' value.
+     *
+     * @link https://www.w3.org/TR/wai-aria-1.1/#aria-expanded
+     */
+    public function ariaExpanded(bool $value = true): self
+    {
+        $new = clone $this;
+        $new->attributes['aria-expanded'] = $value === true ? 'true' : 'false';
+
+        return $new;
+    }
 
-        /** @psalm-suppress InvalidArgument */
-        Html::addCssClass($this->options, ['widget' => 'btn']);
+    /**
+     * Sets the HTML attributes for the button component.
+     *
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function attributes(array $values): self
+    {
+        $new = clone $this;
+        $new->attributes = $values;
 
-        return Html::tag(
-            $this->tagName,
-            $this->encodeLabels ? Html::encode($this->label) : $this->label,
-            $this->options
-        )
-            ->encode($this->encodeTags)
-            ->render();
+        return $new;
     }
 
     /**
-     * When tags Labels HTML should not be encoded.
+     * Sets the 'data-bs-toggle' attribute for the button.
+     *
+     * @param string $value The value to set for the 'data-bs-toggle' attribute.
+     *
+     * @return self A new instance with the specified 'data-bs-toggle' value.
      */
-    public function withoutEncodeLabels(): self
+    public function dataBsToggle(string $value): self
```

Comment:
Are you sure that this method is needed?

By [docs](https://getbootstrap.com/docs/5.3/components/buttons/#toggle-states) this attribute depends from "active" property.

Reply:
Buttons are not only generated for the existing demo, they are also generated as toggles for different components, so they can be used.

Reply:
https://getbootstrap.com/docs/5.2/components/collapse/#example Collapse for example, i can generate the button, and the method is needed.